front/src/applications/stdcm/components/StdcmForm/StdcmLinkedPathSearch.tsx

```diff
@@ -0,0 +1,118 @@
+  className?: string;
+  defaultCardText: string;
+  linkedOp: ExtremityPathStepType;
+};
+
+const StdcmLinkedPathSearch = ({
+  cardIcon,
+  cardName,
+  className = '',
+  defaultCardText,
+  linkedOp,
+}: StdcmLinkedPathSearchProps) => {
+  const { t } = useTranslation('stdcm');
+  const [displayLinkedPathSearch, setShowLinkedPathSearch] = useState(false);
+
+  const {
+    displaySearchButton,
+    hasSearchBeenLaunched,
+    launchTrainScheduleSearch,
+    linkedPathDate,
+    linkedPathResults,
+    selectableSlot,
+    setDisplaySearchButton,
+    setLinkedPathDate,
+    setTrainNameInput,
+    trainNameInput,
+    updateLinkedPathStep,
+  } = useLinkedPathSearch();
+
+  return (
+    <div className={`stdcm-linked-path-search-container ${className}`}>
+      {!displayLinkedPathSearch ? (
+        <div tabIndex={0} role="button" onClick={() => setShowLinkedPathSearch(true)}>
+          <StdcmDefaultCard text={defaultCardText} Icon={cardIcon} className="add-linked-path" />
+        </div>
+      ) : (
+        <StdcmCard
+          name={cardName}
+          title={
+            <button type="button" onClick={() => setShowLinkedPathSearch(false)}>
+              {t('translation:common.delete').toLowerCase()}
+            </button>
+          }
+          className="linked-path"
+        >
+          <div className="d-flex pr-1 pl-3">
+            <Input
+              id="linked-path-id"
+              type="text"
+              value={trainNameInput}
+              onChange={(e) => {
+                setDisplaySearchButton(true);
+                setTrainNameInput(e.target.value);
+              }}
+              label="N°"
+            />
+            <DatePicker
+              inputProps={{
+                id: 'linked-path-date',
+                label: 'Date',
+                name: 'op-date',
+              }}
+              selectableSlot={selectableSlot}
+              value={linkedPathDate}
+              onDateChange={(date) => {
+                setLinkedPathDate(date);
+              }}
+            />
+          </div>
+          {displaySearchButton && (
+            <button
+              className="stdcm-linked-path-button"
+              type="button"
+              onClick={launchTrainScheduleSearch}
+            >
+              {t('find')}
+            </button>
+          )}
+          {!displaySearchButton && !linkedPathResults.length && (
+            <div className="stdcm-linked-path-button white">
+              <Gear size="lg" className="stdcm-linked-path-loading" />
+            </div>
+          )}
+          {linkedPathResults.length > 0 ? (
+            <StdcmLinkedPathResults
+              linkedPathResults={linkedPathResults}
+              onLinkedPathSelection={updateLinkedPathStep}
+              extremityType={linkedOp}
+            />
+          ) : (
+            hasSearchBeenLaunched && (
+              <p className="text-center mb-0">{t('noCorrespondingResults')}</p>
+            )
+          )}
```

Comment:
If you define linkedPathResults like this (undefined by default): 
```
const [linkedPathResults, setLinkedPathResults] = useState<StdcmLinkedPathResult[]>();
```
then you can do:
```suggestion
          {linkedPathResults && (
            <>
              {linkedPathResults.length > 0 ? (
                <StdcmLinkedPathResults
                  linkedPathResults={linkedPathResults}
                  onLinkedPathSelection={updateLinkedPathStep}
                  linkedOp={linkedOp}
                />
              ) : (
                hasSearchBeenLaunched && (
                  <p className="text-center mb-0">{t('noCorrespondingResults')}</p>
                )
              )}
            </>
          )}
```

Reply:
We can do this later :+1: